src/MainWindow.vala

```diff
@@ -347,7 +347,9 @@ public class Dippi.MainWindow : Adw.ApplicationWindow {
             string? text = width_entry.get_text ();
             if (text != null && text != "") {
```

Comment:
It's been a while since I touched this code and I don't really love any of it, _but_... shouldn't this just be handled alongside these other cases?

```suggestion
            if (text != null && text != "" && text != "0") {
```

Reply:
Alternatively if parsing empty and 0 values doesn't crash, I guess we could just parse the text first, and then have the rest inside:

```vala
if (width > 0) { }
```

Reply:
> It's been a while since I touched this code and I don't really love any of it, _but_... shouldn't this just be handled alongside these other cases?

Doing the comparison after the width gets parsed is easier since no matter how many zeros the user types in, the parsed value will always be `0`, making the matching easier

> Alternatively if parsing empty and 0 values returns 0 and doesn't crash, I guess we could just parse the text first, and then have the rest inside:

this would work, and I can't tell you why I didn't do it like that, I assume it's because I saw the current way as "cleaner"

Reply:
That's fair; I feel like cleaning up the logic in the first place is probably better than just sticking an early return in there, since it basically makes the `if` redundant.